graphql-dgs-spring-webflux-autoconfigure/src/main/kotlin/com/netflux/graphql/dgs/webflux/handlers/DgsReactiveWebsocketHandler.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflux.graphql.dgs.webflux.handlers
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.netflix.graphql.dgs.reactive.DgsReactiveQueryExecutor
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscription
+import org.slf4j.LoggerFactory
+import org.springframework.core.ResolvableType
+import org.springframework.core.io.buffer.DataBuffer
+import org.springframework.core.io.buffer.DataBufferUtils
+import org.springframework.http.codec.json.Jackson2JsonDecoder
+import org.springframework.http.codec.json.Jackson2JsonEncoder
+import org.springframework.util.MimeTypeUtils
+import org.springframework.web.reactive.socket.WebSocketHandler
+import org.springframework.web.reactive.socket.WebSocketMessage
+import org.springframework.web.reactive.socket.WebSocketSession
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import java.util.concurrent.ConcurrentHashMap
+
+class DgsReactiveWebsocketHandler(private val dgsReactiveQueryExecutor: DgsReactiveQueryExecutor) : WebSocketHandler {
+
+    private val logger = LoggerFactory.getLogger(DgsReactiveQueryExecutor::class.java)
+    private val resolvableType = ResolvableType.forType(OperationMessage::class.java)
+    private val subscriptions = ConcurrentHashMap<String, MutableMap<String, Subscription>>()
+    private val decoder = Jackson2JsonDecoder()
+    private val encoder = Jackson2JsonEncoder()
+
+    companion object {
+        const val GQL_CONNECTION_INIT = "connection_init"
+        const val GQL_CONNECTION_ACK = "connection_ack"
+        const val GQL_START = "start"
+        const val GQL_STOP = "stop"
+        const val GQL_DATA = "data"
+        const val GQL_ERROR = "error"
+        const val GQL_COMPLETE = "complete"
+        const val GQL_CONNECTION_TERMINATE = "connection_terminate"
+    }
+
+    override fun handle(webSocketSession: WebSocketSession): Mono<Void> {
+        return webSocketSession.send(
```

Comment:
I think, it looks really complicated now. It hard to track flow
And also, as I remember from Reactor specification, it not good solution to call `.subscribe()` inside `.doOnXXX`

Mb it's better to separate input and output? And also change output to resulting command and separated publisher.
Like this:
```kotlin
val publisher: FluxProcessor  = EmitterProcessor() // or other new Sinks builder
val input: Mono<OperationMessage> = webSocketSession.receive().flatMap { .... decoder.decode(... } }
val output: Flux<WebSocketMessage> = 
  input.map { 
    when(om.type) {
        GQL_CONNECTION_INIT -> initConnection(om)
        ....
    } 
val sender = webSocketSession.send(Flux.from(publisher))
return Mono.zip(input, output, sender).then()

fun initConnection(om: OperationMessage): Flux<WebSocketMessage> = 
    Flux.just(publisher.offer(om.exctractInitPayload()))
```

But yes... it also going to be too complicated :)
Not in this PR, it require more investigation and discussions